akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -89,6 +89,11 @@ private[remote] class Encoder(
         val outboundEnvelope = grab(in)
         val envelope = bufferPool.acquire()
 
+        headerBuilder.resetMessageFields()
```

Comment:
It was a real bug, I got wrong manifest/serializerId combination. I think the manifest string from the previous message was used for the next message. Something with one compressed the other not. Could probably be fixed by setting both idx/literal in each setter, but it feels safer to reset the builder for each message.